unstable/wlr-layer-shell-unstable-v1.xml

```diff
@@ -92,9 +92,41 @@
         are not affected.
       </description>
     </request>
+
+    <!-- Version 4 additions -->
+
+    <event name="new_output" since="3">
+      <description summary="notify of a new wl_output">
+        This event indicates that one or more new outputs have been
+        created. All clients must send an ack_new_output request in response
+        to this event. If a client wishes to create a new layer surface in
+        response to the new output(s), it must make an ack_new_output request
+        with the provided serial after creating any new layer surfaces using
+        the get_layer_surface request and before the first commit on any of
+        the new surfaces. If the client does not wish to create a new layer
+        surface, it should ack immediately.
+
+        This allows the compositor to wait until new layer surfaces are ready
+        before rendering the first frame of new outputs, avoiding a race that
+        could cause imperfect frames.
+      </description>
+      <arg name="serial" type="uint"/>
+    </event>
+
+    <request name="ack_new_output" since="3">
+      <description summary="ack a new_output event">
+        This request informs the server that following commits on layer surfaces
+        created by the client take the new output(s) into account.
+
+        A client may make multiple ack_new_output requests before committing.
+        The last request made before the commit indicates which new_output
```

Comment:
Sorry if this is a stupid question, but: what does this last sentence refer to? Aren't next commits taking into account all acked outputs?

Reply:
The idea with this paragraph was to parallel the configure/ack behavior of the xdg-shell protocol. However, I think it may be over-complicating things. There are two paths we could take here:

1. State that new layer surfaces created on a new output before acking that new output are associated only with that change of state.
2. In the case that a second new output is created before the first commit on a new layer surface which was created in response to a previous `new_output` event and the client acks the second new output before committing the layer surface, the layer surface then becomes part of the ack sequence for the second new output.

In it's current state this paragraph is, I believe, trying to explain option 2. I'd suggest that we switch this over to option 1 though as it's simpler and I believe more ideal behavior. Creation of a second new output wouldn't prolong the waiting period for the first frame on a prior new output.

Hopefully that all makes sense, it's rather late here and I don't feel like I've done a terribly good job of putting it into words.

Reply:
IMHO 1. makes more sense